### PR TITLE
[debug] Reserve triggers propperly in HwbpManual

### PR DESCRIPTION
**Pull request by en-sc**
https://github.com/riscv-collab/riscv-openocd/pull/1111 introduces a change in OpenOCD behavior: a manual trigger should be manually removed to step/resume from it.
This was not concidered in previous stop-gap solutions (76ff703d9161945354e334afa45c5dfe7680da60 and 8cc4918e904ca009cd85350fadf6f44e91eca13c)

This commit:
1. Determines if `reserve trigger` is supported by the target. This can be removed once https://github.com/riscv-collab/riscv-openocd/pull/1111 is merged.
2. Marks `HwbpManual` test as not applicable in case `reserve trigger` is not supported.
3. Accounts for the change in OpenOCD's behavior when stepping from a manual BP.
4. Cleans up some minor mistakes in `HwbpManual`

### CODE REVIEW

File: debug/gdbserver.py [hunk 1]
```diff
@@ -728,14 +760,26 @@ def test(self):
...
+            assertNotEqual(before, after)
 
         # Remove the manual HW breakpoint.
         assertEqual(tselect, self.gdb.p("$tselect"))
         self.gdb.p("$tdata1=0")
 
+        if self.reserve_trigger_supported:
+            self.gdb.stepi()
+            after = self.gdb.p("$pc")
+            assertNotEqual(before, after)
```
Comment (aap-sc) [inline debug/gdbserver.py:779]: please, consider a short annotation. for these asserts where you think we may provide some useful context.

File: debug/gdbserver.py [hunk 2]
```diff
@@ -672,10 +672,56 @@ def set_field(reg, mask, val):
...
+
+    reserve_trigger_supported = None
+
     def early_applicable(self):
         return self.target.support_manual_hwbp and \
             self.hart.instruction_hardware_breakpoint_count >= 1
 
+    def check_reserve_trigger_support(self):
+        not_supp_msg = "RESERVE_TRIGGER_NOT_SUPPORTED"
+        self.reserve_trigger_supported = not_supp_msg not in self.gdb.command(
```
[the remark this answers is not in the document]
Comment (en-sc) [reply debug/gdbserver.py]: Addressed.

File: debug/gdbserver.py (hunk 1 above)
Comment (en-sc) [reply debug/gdbserver.py]: Addressed.

File: debug/gdbserver.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (aap-sc) [reply debug/gdbserver.py]: thanks

**Review by aap-sc — APPROVED** [review]

**Pull request merged**